packages/vscode-extension/src/devices/IosSimulatorDevice.ts

```diff
@@ -250,6 +250,23 @@ export class IosSimulatorDevice extends DeviceBase {
     ]);
   }
 
+  async resetAppPermissions(appPermission: AppPermissionType, build: BuildResult) {
+    if (build.platform !== Platform.IOS) {
+      throw new Error("Invalid platform");
+    }
```

Comment:
Why we need this check?

Reply:
to narrow down the type of build from BuildResult to IosBuildResult which has some fields that we need